keep.tex

```diff
@@ -530,38 +531,188 @@ \subsection{Concerns with active attacks}
 
 \section{High-level network design}
 
-To bring these components together into a functional network,
-initially built against the Ethereum blockchain, we'll introduce a few
-more ideas.
+Deploying sMPC-based privacy on a public blockchain requires
+supporting infrastructure. To build a functional privacy network
+against Ethereum, our first target blockchain, we'll introduce
+components to ensure fair keep node selection, report results, and
+incentivize network actors.
 
-\subsection{The keep market}
+\subsection{The Keep network token}
 
-Different keep types will naturally incur different costs, and offer
-differing guarantees and functionality to contract developers. At a
-given time, the capacity available on the keep network will also be
-limited capacity.. For this reason, a market will need to be developed
-that matches keep providers to contracts based on bids for price per
-block, price per operation, and required functionality.
+The native network token, \textit{KEEP}, will be required for
+providers to participate.
 
-\subsection{The Keep network token}
+To be chosen to provide a node for a new keep, a provider must lock up
+a minimum stake in KEEP tokens, using a shared staking contract.
+
+At any time, a provider can choose to retrieve their stake--- for
+example, to liquidate their position. All withdrawals, however, will
+be subject to a two-week waiting period to disincentive providers from
+quickly staking and withdrawing their position, which could have
+adverse effects on running keeps and fair keep selection.
+
+Requiring a native token, rather than the underlying blockchain's
+currency, means providers will suffer from negative externalities in
+the presence of malicious behavior \ref{activeAttacks}. This sort of
+staking also strengthens the system against Sybil attacks
+\ref{fairKeepSelection}.
+
+\subsection{Ensuring fair keep selection}
+\label{fairKeepSelection}
+
+Initially, the network was meant to include an on-chain marketplace to
```

Comment:
Why does it make sense to address a potential previous idea when most readers will not know that this was initially considered? Wouldn't it make more sense to address Sybil Attacks as a whole, and suggest the RNG as the solution?

Reply:
In my mind, the market is the naive solution to the problem. This solution doesn't yet have an answer for other things a market provides, like price discovery.

So I guess the answer is it's a good analogy, and we need to find a way to get the benefits of a market without having a market. I'm going to clarify this language and work on a solution for price discovery and capacity planning. Initial thoughts are doing something inspired by PoW's difficulty retargeting

Reply:
Thanks for pointing that out, this is an awkward way to kick off the topic

Reply:
I do like that this idea was considered as folks may come out of the woodworks and annoy with "what about marketplaces". That being said, there may be a better way to phrase this (as @clp16 is hinting at).

Reply:
Agreed, addressing this